x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/action/TransportGetEnrichPolicyAction.java

```diff
@@ -56,11 +58,18 @@ protected String executor() {
     protected void masterOperation(Task task, GetEnrichPolicyAction.Request request,
                                    ClusterState state,
                                    ActionListener<GetEnrichPolicyAction.Response> listener) throws Exception {
-        final EnrichPolicy policy = EnrichStore.getPolicy(request.getName(), state);
-        if (policy == null) {
-            throw new ResourceNotFoundException("Policy [{}] was not found", request.getName());
+        Map<String, EnrichPolicy> policies;
+        if (request.getName() == null || request.getName().isEmpty()) {
+            policies = EnrichStore.getPolicies(state);
+        } else {
+            EnrichPolicy policy = EnrichStore.getPolicy(request.getName(), state);
+            if (policy == null) {
+                throw new ResourceNotFoundException("Policy [{}] was not found", request.getName());
+            }
+            policies = Collections.singletonMap(request.getName(), policy);
```

Comment:
Nit: maybe use Map.of()? When backporting this change, this then has to be changed back. I just prefer Map.of over Collections.singletonMap in master.

Reply:
I think this is a better solution. I forget about `Map.of`!